### DISCUSSION TITLE
Non-zero mean exact GP with explicit basis functions

### DISCUSSION THREAD

**Discussion by PatLT** [cornellius-gp/gpytorch#1931] [Q&A] [answered]
Hiya, I wanted to have a go at fitting an exact GP with a non-zero mean modelled by explicit basis functions (basically the model in section 2.7 of Rasmussen & Williams). The obvious way to do this would be to have the mean function and kernel take different subsets of the input dimensions. All the kernels have an `active_dims` arg to do just that, but it seems that `mean.LinearMean` doesn't. Is there a way to achieve something similar with existing classes, or will I need to write a custom mean function subclass?

**Comment by wjmaddox** [comment, marked as answer]
Yes, good point, adding active dims to linear mean would be a good idea for a pull request. 

One other thing you could try is to do the sub-setting automatically in the forwards call:
```python
def forward(self, x):
    mean = self.mean_module(x[..., :3]) # the mean only takes as inputs the first two dimensions
    covar = self.mean_module(x)
    return MultivariateNormal(mean, covar)
```
